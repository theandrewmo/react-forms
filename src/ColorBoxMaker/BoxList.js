import { useState } from 'react';
import Box from './Box'
import NewBoxForm from './NewBoxForm';
import {v4 as uuid} from 'uuid';

/** BoxList - renders all of the boxes along with the new box form
 * 
 * Props: none
 * 
 * State: array of all boxes
 * 
 */

const BoxList = () => {

    const [boxes, setBoxes] = useState([])

    const removeBox = (key) => {
        setBoxes(boxes.filter(box => (box.key !== key)))
    }

    const addBox = (backgroundColor, width, height) => {
        const boxKey = uuid();
        const newBox = {
            key: boxKey,
            backgroundColor: backgroundColor,
            width: width,
            height: height,
          };
        setBoxes([...boxes, newBox])
    }
    return (
        <>
            {boxes.map((box) => (
                <div key={box.key}>
                <Box
                    backgroundColor={box.backgroundColor}
                    width={box.width}
                    height={box.height}
                    removeBox={() => removeBox(box.key)}
                />
                </div>
             ))}

            < NewBoxForm addBox={addBox}/>
        </>
    )
}

export default BoxList;