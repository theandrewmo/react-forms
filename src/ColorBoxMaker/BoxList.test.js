import { render, screen, fireEvent} from '@testing-library/react';
import BoxList from './BoxList';

it('renders', () => {
  render(<BoxList />);
});

it("matches snapshot", () => {
    const { asFragment } = render(<BoxList
        />);
    expect(asFragment()).toMatchSnapshot();
})

it('Adds a new box', () => {
  render(<BoxList />);

  const backgroundColorInput = screen.getByTestId('backgroundColor');
  const widthInput = screen.getByTestId('width');
  const heightInput = screen.getByTestId('height');
  const submitButton = screen.getByTestId('submitButton'); 

  // Set values for the input fields
  fireEvent.change(backgroundColorInput, { target: { value: 'red' } });
  fireEvent.change(widthInput, { target: { value: '100px' } });
  fireEvent.change(heightInput, { target: { value: '100px' } });

  // Click the submit button
  fireEvent.click(submitButton);

  // Assert that the new box is added based on its properties
  expect(screen.getByText('X')).toBeInTheDocument(); // Check for the delete button
  expect(screen.getByTestId("box-red-100px-100px")).toBeInTheDocument(); // Check for the box content
  expect(screen.getByTestId("box-red-100px-100px")).toHaveStyle('background-color: red'); // Check for the box background color
  expect(screen.getByTestId("box-red-100px-100px")).toHaveStyle('width: 100px'); // Check for the box width
  expect(screen.getByTestId("box-red-100px-100px")).toHaveStyle('height: 100px'); // Check for the box height
});