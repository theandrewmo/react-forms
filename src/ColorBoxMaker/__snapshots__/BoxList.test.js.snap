// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="backgroundColor"
    >
      Background Color: 
    </label>
    <input
      data-testid="backgroundColor"
      id="backgroundColor"
      name="backgroundColor"
      placeholder="backgroundColor"
      value=""
    />
    <label
      for="width"
    >
      Width: 
    </label>
    <input
      data-testid="width"
      id="width"
      name="width"
      placeholder="width"
      value=""
    />
    <label
      for="height"
    >
      Height: 
    </label>
    <input
      data-testid="height"
      id="height"
      name="height"
      placeholder="height"
      value=""
    />
    <button
      data-testid="submitButton"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
